api/v1alpha1/database_types.go

```diff
@@ -84,25 +84,24 @@ type NDB struct {
 // Database instance specific details
 type Instance struct {
 	// Name of the database instance
-	DatabaseInstanceName string `json:"databaseInstanceName"`
+	DatabaseInstanceName *string `json:"databaseInstanceName"`
 	// Name(s) of the database(s) to be provisiond inside the database instance
 	// default [ "database_one", "database_two", "database_three" ]
 	// +optional
-	DatabaseNames []string `json:"databaseNames"`
+	DatabaseNames *[]string `json:"databaseNames"`
 	// Name of the secret holding the credentials for the database instance (password and ssh key)
-	CredentialSecret string `json:"credentialSecret"`
+	CredentialSecret *string `json:"credentialSecret"`
 	// Size of the database instance, minimum 10 (GBs)
-	Size int `json:"size"`
+	Size *int `json:"size"`
 	// default UTC
 	// +optional
-	TimeZone string `json:"timezone"`
-	Type     string `json:"type"`
+	TimeZone *string `json:"timezone"`
+	Type     *string `json:"type"`
 	// +optional
-	Profiles Profiles `json:"profiles"`
+	Profiles *Profiles `json:"profiles"`
 	// +optional
-	// +kubebuilder:default:="{emptytminfo}"
 	// Information related to time machine that is to be associated with this database
-	TMInfo DBTimeMachineInfo `json:"timeMachine"`
+	TMInfo *DBTimeMachineInfo `json:"timeMachine"`
 }
```

Comment:
As discussed in the Friday sync-up call, we just need to mark the top-level fields right now. This is because depending upon clone/new provisioning, a certain set of inputs are needed or not needed, but I am not immediately be able to see where a partial set is needed. If we come across such a case, I will convert those into pointers too.

As of now, trying not to over-optimize in anticipation.